pkg/solana/codec/byte_string_modifier.go

```diff
@@ -0,0 +1,42 @@
+package codec
+
+import (
+	"fmt"
+
+	"github.com/gagliardetto/solana-go"
+
+	commontypes "github.com/smartcontractkit/chainlink-common/pkg/types"
+)
+
+// SolanaAddressModifier implements the AddressModifier interface for Solana addresses.
+// It handles encoding and decoding Solana addresses using Base58 encoding.
+type SolanaAddressModifier struct{}
+
+// EncodeAddress encodes a Solana address (32-byte array) into a Base58 string.
+func (s SolanaAddressModifier) EncodeAddress(bytes []byte) (string, error) {
+	if len(bytes) != s.Length() {
+		return "", fmt.Errorf("%w: got length %d, expected 32 for bytes %x", commontypes.ErrInvalidType, len(bytes), bytes)
+	}
+	return solana.PublicKeyFromBytes(bytes).String(), nil
+}
+
+// DecodeAddress decodes a Base58-encoded Solana address into a 32-byte array.
+func (s SolanaAddressModifier) DecodeAddress(str string) ([]byte, error) {
+	// Validate the length of the address
```

Comment:
comment the why, not the what, a lot of the comments in this file can be very easily derived from the code